tests/lax_test.py

```diff
@@ -161,6 +162,21 @@ def op_record(op, nargs, dtypes, rng_factory, tol=None):
 class LaxTest(jtu.JaxTestCase):
   """Numerical tests for LAX operations."""
 
+  def _CheckAgainstNumpy(self, numpy_reference_op, lax_op, args_maker,
+                         check_dtypes=True, tol=None, canonicalize_dtypes=True):
+    """Override base method to also check against numpy_eval."""
+    args = args_maker()
+    lax_ans = lax_op(*args)
+    numpy_ans = numpy_reference_op(*args)
+    self.assertAllClose(numpy_ans, lax_ans, check_dtypes=check_dtypes,
+                        atol=tol, rtol=tol,
+                        canonicalize_dtypes=canonicalize_dtypes)
+    if config.omnistaging_enabled:
```

Comment:
The slightly strange part of putting this in `CheckAgainstNumpy` is that it should actually be a well-defined check even when you don't have a `numpy_reference_op`. That's why I suggested putting it in `CompileAndCheck` instead, or at least a version of that used by `lax_test.py` specifically which is where it would be most important as a test.

Reply:
Ah, great point! Did that. This washed up a bunch of bugs in the numpy implementation. I fixed all except one in conv, left a TODO there.